src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IInternalStringEqualityComparer.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Represents an <see cref="IEqualityComparer{String}"/> that's meant for internal
+    /// use only and isn't intended to be serialized or returned back to the user.
+    /// Use the <see cref="GetUnderlyingEqualityComparer"/> method to get the object
+    /// that should actually be returned to the caller.
+    /// </summary>
+    internal interface IInternalStringEqualityComparer : IEqualityComparer<string?>
+    {
+        IEqualityComparer<string?> GetUnderlyingEqualityComparer();
+    }
+
+    internal static class InternalStringEqualityComparer
```

Comment:
Nit: The static method can be in `IInternalStringEqualityComparer`. No need to have a separate type with just a single method.

Reply:
Do we have other examples of where we've introduced static methods in interfaces?

Reply:
We do not have much, if any. It got enabled in C# not too long ago. C# allows it now so we should be free to use it. Using it as internal implementation detail is low risk - we can undo it if necessary.